dateparser/custom_language_detection/fasttext.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+import fasttext
+
+from dateparser_cli.fasttext_manager import fasttext_downloader
+from dateparser_cli.utils import dateparser_model_home, create_data_model_home
+from dateparser_cli.exceptions import FastTextModelNotFoundException
+
+
+_supported_models = ["large.bin", "small.bin"]
+_DEFAULT_MODEL = "small"
+
+
+class _FastTextCache:
+    model = None
+
+
+def _load_fasttext_model():
+    if _FastTextCache.model:
+        return _FastTextCache.model
+    create_data_model_home()
+    model_path = None
+    downloaded_models = os.listdir(dateparser_model_home)
+    for downloaded_model in _supported_models:
+        if downloaded_model in downloaded_models:
+            model_path = os.path.join(dateparser_model_home, downloaded_model)
+            break
+    if not model_path:
+        fasttext_downloader(_DEFAULT_MODEL)
+        return _load_fasttext_model()
```

Comment:
```suggestion
    downloaded_models = [
        file for file in os.listdir(dateparser_model_home)
        if file in _supported_models
    ]
    if not downloaded_models:
        fasttext_downloader(_DEFAULT_MODEL)
        return _load_fasttext_model()

    model_path = os.path.join(dateparser_model_home, downloaded_models[0])
```

I think this approach is better.

Reply:
Thanks Marc, changes done 👍🏻